pytalibstream/atr_keeper.py

```diff
@@ -0,0 +1,41 @@
+import pyslidingwindow
+
+
+class AtrKeeper:
+    def __init__(self, max_len: int):
+        if max_len < 2:
+            raise Exception("AtrKeeper period must be >= 2")
+
+        self.high = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.low = pyslidingwindow.SlidingWindowFloat(max_len)
+        self.close = pyslidingwindow.SlidingWindowFloat(max_len)
+        self._atr = 0
+        self.data_count = 0
+        self.max_len = max_len
+        self.prev_tr = []
+
+    def add(self, high: float, low: float, close: float):
+        self.data_count += 1
+        self.high.push(high)
+        self.low.push(low)
+        self.close.push(close)
+
+        if self.max_len + 1 > self.data_count > 1:
+            self.prev_tr.append(self.calc_tr())
```

Comment:
in the nodejs code, nested if is necessary. it implicitly says `if this.dataLen  == 1` no nothing
```
if (this.dataLen < this.period + 1) {
    // if this.dataLen  == 1 no nothing
      if (this.dataLen > 1) {
        this.previousTr.push(this.getTr());
      }
    }
```

Reply:
if you don't have this nested if, would the `this.dataLen  == 1` case go into the else scenario?

Reply:
Yes. It will check the elif condition first but ultimately it will go into the else case

Reply:
sorry, I think there are some disconnection here.
when `this.dataLen == 1` it should not go to the else case. it should do nothing.

Reply:
Oh I see what you mean, Ill make it so that it never goes past that first if in that case